Multitrack-Template.lua

```diff
@@ -0,0 +1,205 @@
+--[[
+Multi-track re-record script template.
+
+This template was created from DarkKobold's Multitrack LUA script created
+for TAS of TMNT Arcade and The Simpsons Arcade.
+
+The intent of this template script is to make multiplayer TAS'ing more
+approachable to newbies - empowering them to quickly create a
+multi-track re-recording rig for the game they plan to TAS
+--]]
+
+-- Max number of lines to display in LUA overlay
+MAX_DISPLAY_LINES = 5;
+
+-- local NameTable = {'Player 1', 'Player 2', 'None', 'All'};
+local NameTable = {'Player 1', 'Player 2'};
+local visualtbl = {'^','v','<','>','O','O','O'}
+local tabinp = {'up','down','left','right','Z','X','C','9','0'};
+
+-- Dictionary holding Key Value Pairs
+-- for input labels
+-- Key is derived from I don't know what yet
+local i = {};
+for n=(PLAYER_COUNT-1),0,-1 do
+	i[1+n*INPUT_BUTTON_COUNT] = "P"..(n+1).." Up";
+	i[2+n*INPUT_BUTTON_COUNT] = "P"..(n+1).." Down";
+	i[3+n*INPUT_BUTTON_COUNT] = "P"..(n+1).." Left";
+	i[4+n*INPUT_BUTTON_COUNT] = "P"..(n+1).." Right";
+	i[5+n*INPUT_BUTTON_COUNT] = "P"..(n+1).." Button 1";
+	i[6+n*INPUT_BUTTON_COUNT] = "P"..(n+1).." Button 2";
+	i[7+n*INPUT_BUTTON_COUNT] = "P"..(n+1).." Button 3";
+	i[8+n*INPUT_BUTTON_COUNT] = "P"..(n+1).." Start";
+	i[9+n*INPUT_BUTTON_COUNT] = "P"..(n+1).." Coin";
+end
+
+-- Max number of players
+PLAYER_COUNT = table.getn(NameTable);
+
+-- Total number of buttons being drawn in UI
+MONITORED_BUTTON_COUNT = table.getn(visualtbl);
+
+-- Table representing all the input buttons for your game
+INPUT_BUTTON_COUNT = table.getn(tabinp);
+
+current_pressed_keys, last_pressed_keys = {}, {};
+AllFrames = {};
+
+local inp ={};
+local StartFrame = emu.framecount();
+local RecFrames = 0;
+
+-- This is the character whose inputs are currently being recorded
```

Comment:
heads up - some of the comments aren't correct [I was writing notes to myself as I get better idea of what the script is doing]